clients/src/main/java/org/apache/kafka/clients/admin/QuorumInfo.java

```diff
@@ -29,19 +32,22 @@ public class QuorumInfo {
     private final long highWatermark;
     private final List<ReplicaState> voters;
     private final List<ReplicaState> observers;
+    private final Map<Integer, Node> nodes;
```

Comment:
Could you explain why we need a map here , not a Set<Node> ? It looks like we already store nodeId in Node object, and we don't need the nodeId as key, right?

Reply:
> It looks like we already store nodeId in Node object, and we don't need the nodeId as key, right?

Correct. My first implementation was Set based. But, @cmccabe ask to refactor to Map
See - https://github.com/apache/kafka/pull/16106#discussion_r1628472921

Reply:
It's easier for users this way. The extra int is not a major burden since the map is so small.